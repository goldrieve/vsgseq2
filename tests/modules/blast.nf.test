nextflow_process {

    name "Test Process BLAST"
    script "modules/blast.nf"
    process "BLAST"

    test("Should blast ORFs and output VSGs") {

        when {
            params {
                vsg_db = "$projectDir/data/blastdb/concatAnTattb427.fa"
                notvsg_db = "$projectDir/data/blastdb/vsgseq2NOTvsgs.fa"
                scripts = "$projectDir/bin/"
            }
            process {
                """
                input [0] = "$projectDir/tests/data/latesingle_trinity_cdhit.fasta"
                input [1] = params.vsg_db
                input [2] = params.notvsg_db 
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out.vsgs).match()
        }

    }

}
