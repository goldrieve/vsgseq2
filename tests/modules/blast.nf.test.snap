{
    "Should run without failures": {
        "content": [
            [
                "latesingle_trinity_cdhit_VSGs.fasta:md5,966afb15c69677a4245f6b30f44ef008"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-08-30T13:19:32.764113355"
    }
}